### PR TITLE
Support for dynamic network config

### PR DESCRIPTION
**Pull request by chenchun**
- Load unknown network config from /etc/cni/net.d/ to co-work with other
cni plugins
- Mount cni binary path to the same path in container so galaxy may find
binaries of other cnis

### CODE REVIEW

**Comment by chenchun** [comment]
Fixes https://github.com/tkestack/galaxy/issues/17

**Comment by gongguan** [comment]
These change looks good to me.

File: pkg/galaxy/galaxy.go [hunk 1]
```diff
@@ -86,26 +86,33 @@ func (g *Galaxy) Init() error {
...
 		} else {
-			g.netConf[netType] = g.NetworkConf[i]
+			// name empty, assume type name is network name
+			key = netType
 		}
-	}
-	for _, netType := range g.DefaultNetworks {
-		if _, ok := g.netConf[netType]; !ok {
-			return fmt.Errorf("network configuration is empty for default network %s", netType)
+		if _, ok := g.netConf[key]; ok {
```
[the remark this answers is not in the document]
Comment (chenchun) [reply pkg/galaxy/galaxy.go]: ok is expected.  Network `name` expects to be uniq.

File: pkg/galaxy/server.go [hunk 2]
```diff
@@ -218,6 +221,32 @@ func (g *Galaxy) resolveNetworks(req *galaxyapi.PodRequest, pod *corev1.Pod) ([]
...
+		return nil
+	}
+	var m map[string]interface{}
+	if err := json.Unmarshal(data, &m); err != nil {
+		glog.Warningf("failed to unmarshal networkinfo %s: %v", string(data), err)
+		return nil
+	}
+	// kubeconfig of host filesystem won't be reachable for galaxy.
+	// Since galaxy is running in pod, it can talk to apiserver via secret token
+	if m["kubeconfig"] != "" {
```
Comment (ChenLingPeng) [inline pkg/galaxy/server.go]: kubeconfig is useful anywhere else? if not, we should assume where's no kubeconfig entry, or when init k8sClient, kubeconfig entry should not be readed, so the if statement could be removed

Comment (chenchun) [reply pkg/galaxy/server.go]: Galaxy doesn't depend on `kubeconfig`, you can check [galaxy-etc configmap](https://github.com/tkestack/galaxy/blob/master/yaml/galaxy.yaml#L149-L178).
This is to ensure other cnis which depend on kubeconfig to talk to apiserver to be continue working via service token within galaxy pod.

**Pull request merged**